codegen/lib/graphql_java_gen.rb

```diff
@@ -253,10 +253,15 @@ def java_implements(type)
     "implements #{interfaces.to_a.join(', ')} "
   end
 
-  def java_annotations(field)
-    @annotations.map do |annotation|
+  def java_annotations(field, in_argument: false)
+    annotations = @annotations.map do |annotation|
       "@#{annotation.name}" if annotation.annotate?(field)
-    end.compact.join("\n")
+    end.compact
+    if in_argument
+      annotations.join(" ") + " "
```

Comment:
There's still the problem that we need a trailing space at the end of annotations in argument that I feel would make the templates uglier to deal with it there. I'll just add an early return if there are no annotations